src/Database/PostgreSQL/Transact/Cluster.hs

```diff
@@ -152,7 +149,7 @@ readonly = CDBT
 -- | Mark any query as a 'ReadWrite'.  This is useful for including 'ReadOnly'
 -- code in a 'ReadWrite' value
 asReadWrite :: CDBT mode m a -> CDBT 'ReadWrite m a
-asReadWrite = CDBT . getDBT
+asReadWrite = coerce
```

Comment:
This isn't absolutely necessary, but it does enforce that this `asReadWrite` function has no cost.